From 94dd2c488d4da5e18a3767da7b22cb501fcd621b Mon Sep 17 00:00:00 2001
From: Antoine Tenart <antoine.tenart@bootlin.com>
Date: Wed, 13 Feb 2019 16:40:51 +0100
Subject: [PATCH 105/386] net: mvpp2: reconfiguring the port interface is
 PPv2.1 specific

This patch adds a check not to disable/enable a port when an interface
is updated when using PPv2.1 as the functions called are PPv2.2
specific.

Change-Id: I067ea35c399313df10ca324f3c71bcec3a4174c4
Signed-off-by: Antoine Tenart <antoine.tenart@bootlin.com>
Reviewed-on: https://sj1git1.cavium.com/6473
Tested-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 15 ++++++---------
 1 file changed, 6 insertions(+), 9 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 6d3f5f618f1d..68b97551a597 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -6137,21 +6137,18 @@ static void mvpp2_mac_config(struct net_device *dev, unsigned int mode,
 		break;
 	};
 
-	/* Mac-config is called on UP-request. Don't repeat if already UP */
-	if (change_interface) {
+	if (port->priv->hw_version != MVPP21 && change_interface) {
 		/* Make sure the port is disabled when reconfiguring the mode */
 		mvpp2_tx_stop_all_queues(port->dev);
 		mvpp2_port_disable(port);
 
 		mvpp22_gop_mask_irq(port);
 
-		if (port->priv->hw_version == MVPP22) {
-			port->phy_interface = state->interface;
+		port->phy_interface = state->interface;
 
-			/* Reconfigure the serdes lanes */
-			phy_power_off(port->comphy);
-			mvpp22_mode_reconfigure(port);
-		}
+		/* Reconfigure the serdes lanes */
+		phy_power_off(port->comphy);
+		mvpp22_mode_reconfigure(port);
 
 		mvpp2_tx_wake_all_queues(dev);
 		mvpp2_port_enable(port);
@@ -6171,7 +6168,7 @@ static void mvpp2_mac_config(struct net_device *dev, unsigned int mode,
 	if (port->priv->hw_version == MVPP21 && port->flags & MVPP2_F_LOOPBACK)
 		mvpp2_port_loopback_set(port, state);
 
-	if (change_interface)
+	if (port->priv->hw_version != MVPP21 && change_interface)
 		mvpp22_gop_unmask_irq(port);
 }
 
-- 
2.17.1

